include/wil/cppwinrt_register_com_server.h

```diff
@@ -0,0 +1,83 @@
+//*********************************************************
+//
+//    Copyright (c) Microsoft. All rights reserved.
+//    This code is licensed under the MIT License.
+//    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF
+//    ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
+//    TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+//    PARTICULAR PURPOSE AND NONINFRINGEMENT.
+//
+//*********************************************************
+//! @file
+//! Utilities for making managing COM server easier
+
+#ifndef __WIL_CPPWINRT_REGISTER_COM_SERVER_INCLUDED
+#define __WIL_CPPWINRT_REGISTER_COM_SERVER_INCLUDED
+
+#include <winrt/base.h>
+#include <wil/resource.h>
+#include <vector>
+
+namespace wil::details
+{
+template <typename T>
+struct CppWinRTClassFactory : winrt::implements<CppWinRTClassFactory<T>, IClassFactory, winrt::no_module_lock>
+{
+    HRESULT __stdcall CreateInstance(IUnknown* outer, GUID const& iid, void** result) noexcept final
+    {
+        *result = nullptr;
+
+        if (outer)
+        {
+            return CLASS_E_NOAGGREGATION;
+        }
+
+        return winrt::make_self<T>().as(iid, result);

```

Comment:
Should this have function-level `try...CATCH_RETURN()`?

Reply:
Added. Also handled `winrt::hresult_error` as that caused `CATCH_RETURN` to fail fast.

Reply:
Oh right, that's probably because `wil/cppwinrt.h` isn't getting included. This should be fine.

Reply:
Updated to use `wil/cppwinrt.h` :D